overlord/devicestate/devicestate.go

```diff
@@ -541,6 +536,10 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, pathSI *pat
 	}
 
 	if !changed {
+		addExistingSnapTasks := snapstate.LinkNewBaseOrKernel
```

Comment:
I like this change, but I think it'd be even more clear if `addExistingSnapTasks` was eliminated and the functions were called directly.